src/db.c

```diff
@@ -1246,34 +1246,68 @@ void scanGenericCommand(client *c, robj *o, unsigned long long cursor) {
         unsigned char *p = lpFirst(o->ptr);
         unsigned char *str;
         int64_t len;
+        unsigned long array_reply_len = 0;
         unsigned char intbuf[LP_INTBUF_SIZE];
-
+        void *replylen = NULL;
+        listRelease(keys);
+
+        /* Reply to the client. */
+        addReplyArrayLen(c, 2);
+        /* Cursor is always 0 given we iterate over all set */
+        addReplyBulkLongLong(c,0);
+        /* If there is no pattern the length is the entire set size, otherwise we defer the reply size */
+        if (use_pattern)
+            replylen = addReplyDeferredLen(c);
+        else {
+            array_reply_len = o->type == OBJ_HASH ? hashTypeLength(o, 0) : zsetLength(o);
+            if (!no_values){
+                array_reply_len *= 2;
```

Comment:
i'm worried it will be overflow if the size more than ULONG_MAX / 2.

Reply:
@sundb i'm not sure if the check is really needed given the following.
by default hash-max-listpack-entries (512) will limit it to 1024. Nonetheless, even if the user changes it to max 2147483647 our reply whould be 2x that size, which means 2x LONG_MAX =  4294967294, which is lower than the max 4294967295.

Reply:
you forget the corrupt data.

Reply:
@sundb for th overflow corrupted data, it's not possible to go over it even with corrupted data given the listpack header size for number of elements is just 2 bytes long https://github.com/redis/redis/blob/unstable/src/listpack.c#L26C80-L27C20 
```
+----------+-------------+-------------------+----------+
| Offsets  | Size        | Description       | Notes    |
+----------+-------------+-------------------+----------+
| 0 - 3    | 4 bytes     | total_bytes       | uint32_t |
| 4 - 5    | 2 bytes     | num_elements      | uint16_t |
| 6 - ...  | Variable    | Entries           |          |
| ...      | 1 byte      | End byte (0xFF)   |          |
+----------+-------------+-------------------+----------+
```

Reply:
@fcostaoliveira yeah, you're right.